### PR TITLE
[Train/docs] Deduplicate Train examples & move from `ray.air` to `ray.train`

### LINKED ISSUE #26536: [AIR/Docs] Deduplicate AIR-Train examples
### Description

Legacy Train examples have been converted to use Train. Some of those examples showcase the same features or are outright copies of examples created specifically for AIR. We should review the examples and remove unnecessary ones.

Duplicates:
- python/ray/train/examples/torch_fashion_mnist_example.py : python/ray/air/examples/pytorch/torch_fashion_mnist_example.py
- python/ray/train/examples/torch_linear_example.py : python/ray/air/examples/pytorch/torch_linear_example.py
- python/ray/train/examples/tensorflow_mnist_example.py : python/ray/air/examples/tf/tensorflow_mnist_example.py
- python/ray/train/examples/tune_tensorflow_mnist_example.py : python/ray/air/examples/tf/tune_tensorflow_mnist_example.py

### Use case

_No response_

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

This PR moves Train examples from `ray.air.examples` to `ray.train.examples`, making sure that duplicate examples are removed and the folder structure is preserved.

## Related issue number

Closes https://github.com/ray-project/ray/issues/26536

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/BUILD [hunk 1]
```diff
@@ -71,23 +71,154 @@ py_test(
...
+    size = "small",
+    main = "examples/pytorch_geometric/distributed_sage_example.py",
+    srcs = ["examples/pytorch_geometric/distributed_sage_example.py"],
+    tags = ["team:ml", "exclusive", "gpu"],
+    deps = [":train_lib"],
+    args = ["--use-gpu", "--num-workers=2", "--epochs=1", "--dataset=fake"]
+)
+
+py_test(
+    name = "horovod_cifar_pbt_example",
```
Comment (amogkam) [inline python/ray/train/BUILD:104]: do we need an `__init__.py` in the ray/train/examples/horovod?

File: python/ray/train/examples/mlflow_fashion_mnist_example.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import argparse
 
 from ray.air.config import RunConfig, ScalingConfig
-from ray.train.examples.torch_fashion_mnist_example import train_func
+from ray.train.examples.pytorch.torch_fashion_mnist_example import train_func
 from ray.train.torch import TorchTrainer
 from ray.air.callbacks.mlflow import MLflowLoggerCallback
```
Comment (amogkam) [inline python/ray/train/examples/mlflow_fashion_mnist_example.py:6]: should this example be moved into `pytorch`?

**Review by amogkam — COMMENTED** [review]
thanks, lgtm! just some minor comments

File: python/ray/train/BUILD [hunk 3]
```diff
@@ -71,23 +71,154 @@ py_test(
...
+#     name = "torch_fashion_mnist_example",
+#     size = "medium",
+#     main = "examples/pytorch/torch_fashion_mnist_example.py",
+#     srcs = ["examples/pytorch/torch_fashion_mnist_example.py"],
+#     tags = ["team:ml", "exclusive"],
+#     deps = [":train_lib"],
+#     args = ["--smoke-test"]
+# )
+
+# py_test(
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/train/BUILD:181]: `test_gpu_examples`, will add a comment

File: python/ray/train/BUILD [hunk 4]
```diff
@@ -71,23 +71,154 @@ py_test(
...
+#     main = "examples/tf/tensorflow_mnist_example.py",
+#     srcs = ["examples/tf/tensorflow_mnist_example.py"],
+#     tags = ["team:ml", "exclusive"],
+#     deps = [":train_lib"],
+#     args = ["--smoke-test"]
+# )
+
+# py_test(
+#     name = "tensorflow_autoencoder_example", # REGRESSION
+#     size = "medium",
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/train/BUILD:162]: Actually, the .py file got moved here. The jupyter notebook is in ray/air, and that is not what this entry is referring to.

File: python/ray/train/BUILD (hunk 1 above)
Comment (Yard1) [reply python/ray/train/BUILD:104]: We may run into issues with imports (and we are importing functions from the examples) without it.

File: python/ray/train/examples/mlflow_fashion_mnist_example.py (hunk 2 above)
Comment (Yard1) [reply python/ray/train/examples/mlflow_fashion_mnist_example.py:6]: I guess it is using pytorch, but on the other hand, it's supposed to show off MLFlow first and foremost and pytorch is only an implementation detail here. I'd just leave it where it is for better visibility.

**Review by amogkam — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Comment by amogkam** [comment]
Failing tests are also failing on master. Going to merge.

**Pull request merged**